### PR TITLE
Allow Validator registration in L2 without BLS key

### PR DESCRIPTION
**Pull request by m-chrzan**
### Description

BLS keys were used in CEL1 for PoS signatures. They're no longer needed in CEL2. 

### Other changes

* Reallowed `registerValidatorGroup`, `updateEcdsaPublicKey` in L2.
* `isValidator` check now checks the ECDSA key rather than the BLS key.
* In MockElection, allow `setAllowedToVoteOverMaxNumberOfGroups` in L2.

### Tested

Unit tests.

### Related issues

- Fixes https://github.com/celo-org/celo-blockchain-planning/issues/556

### Backwards compatibility

Introduces a new function that works only in L2.

### CODE REVIEW

File: packages/protocol/contracts/governance/Validators.sol [hunk 1]
```diff
@@ -1109,7 +1136,7 @@ contract Validators is
    * @return Whether a particular address is a registered validator.
    */
   function isValidator(address account) public view returns (bool) {
-    return validators[account].publicKeys.bls.length > 0;
+    return validators[account].publicKeys.ecdsa.length > 0;
```
Comment (martinvol) [inline packages/protocol/contracts/governance/Validators.sol:1150]: shouldn't this behaviour change after the L2?

File: packages/protocol/test-sol/unit/governance/validators/Validators.t.sol [hunk 2]
```diff
@@ -804,6 +821,191 @@ contract ValidatorsTest_RegisterValidator is ValidatorsTest {
...
+    validators.registerValidator(_registeredEcdsaPubKey);
+  }
+
+  function test_Reverts_WhenAccountAlreadyRegisteredAsValidatorGroup() public {
+    _whenL2();
+    _registerValidatorGroupHelper(validator, 1);
+    vm.expectRevert("Already registered");
+    vm.prank(validator);
+    validators.registerValidator(
+      abi.encodePacked(bytes32(0x0101010101010101010101010101010101010101010101010101010101010101))
```
Comment (soloseng) [inline packages/protocol/test-sol/unit/governance/validators/Validators.t.sol:992]: Any reason for not using the `_generateEcdsaPubKey` here to get the ecdsaPubKey?

**Review by soloseng — APPROVED** [review]
Looks good 🚀

Left just one comment.

File: packages/protocol/contracts/governance/Validators.sol (hunk 1 above)
Comment (m-chrzan) [reply packages/protocol/contracts/governance/Validators.sol:1150]: What do you mean?

File: packages/protocol/test-sol/unit/governance/validators/Validators.t.sol (hunk 2 above)
Comment (m-chrzan) [reply packages/protocol/test-sol/unit/governance/validators/Validators.t.sol:992]: I mostly just copy-pasted the existing tests for the existing, CEL1 version of the function, that's how this test was implemented there.

File: packages/protocol/contracts/governance/Validators.sol (hunk 1 above)
Comment (martinvol) [reply packages/protocol/contracts/governance/Validators.sol:1150]: the check should work as usual before the L1 (when this code is deployed) and then change logic. Am I right?

Comment (m-chrzan) [reply packages/protocol/contracts/governance/Validators.sol:1150]: This check will work in both L1 and L2 - in either case it depends only on checking some value that is non-0 for a registered validator. In fact, in prior versions of this code, it used to check a different part of the struct. So there should be no need to increase complexity with an L2 check. The check is different, but behavior is equivalent.

**Pull request merged**